lwreg/utils.py

```diff
@@ -244,6 +246,66 @@ def _register_mol(tpl, escape, cn, curs, config, failOnDuplicate):
     return mrn
 
 
+def _get_delimiter(smilesfile):
+    with open(smilesfile, 'r') as input_file:
+        lines = input_file.readlines()
+        sniffer = csv.Sniffer()
+        dialect = sniffer.sniff(lines[-1])
+        delimiter = dialect.delimiter
+        expected_delimiters = ['\t', ',', ';', ' ', '  ', '   ', '    ', '|']
+        if delimiter in expected_delimiters:
+            return delimiter
+        else:
+            return
+
+
+def _get_smiles_column(smilesfile, delimiter):
+    with open(smilesfile, 'r', newline='') as input_file:
+        rows = csv.reader(input_file, delimiter=delimiter)
+        rows = list(rows)
+        for row in rows[::-1]:
+            with rdBase.BlockLogs() as bl:
+                potential_smiles = list([type(Chem.MolFromSmiles(smi)) for smi in row])
+            try:
+                smi = potential_smiles.index(Chem.rdchem.Mol)
+                return smi
+            except ValueError:
+                pass
```

Comment:
I think think this is equivalent (not tested), faster,and simpler:
```suggestion
            with rdBase.BlockLogs() as bl:
                for i,smi in enumerate(row):
                    if Chem.MolFromSmiles(smi,sanitize=False) is not None:
                        return i
```
Plus you really should be including `sanitize=False` in this since it may be that people want to start with non-sanitizeable molecules.

Reply:
It looks better this way. I have applied the suggested changes.
[8bfe83d](https://github.com/rinikerlab/lightweight-registration/pull/21/commits/8bfe83dbef1748fd5009c13292a3bf05dfb1ffc3)